src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,42 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Figure figure;
+
+    public Figure getRandomFigure() {
+        int numOfFigure = random.nextInt(Figure.amountOfFigures);
```

Comment:
Magic numbers in your code should be constants. The `Figure.amountOfFigures` should be a constant in the `FigureSupplier` class with a clear name indicating its purpose and use the proper naming convention for constants.